test/services/pilot/Makefile

```diff
@@ -0,0 +1,63 @@
+ROOT_DIR := ../../..
+include $(ROOT_DIR)/Makefile.env
+include $(ROOT_DIR)/.mk/docker.mk
+
+MOCK_DIRS := \
+	catalog-credential-mock \
+	policy-manager-mock
+
+define build-target
+  build:: ; cd $1 && make build
+endef
+
+define clean-target
+  clean:: ; cd $1 && make clean
+endef
+
+define test-target
+  test:: ; cd $1 && make test
+endef
+
+define run-target
+  run:: ; cd $1 && make run
+endef
+
+define terminate-target
+  terminate:: ; cd $1 && make terminate
+endef
+
+define docker-all-target
+  dockerall:: ; cd $1 && make docker-all
+endef
+
+define docker-build-target
+  dockerbuild:: ; cd $1 && make docker-build
+endef
+
+define docker-push-target
+  dockerpush:: ; cd $1 && make docker-push
+endef
+
+define docker-rmi-target
+  dockerrmi:: ; cd $1 && make docker-rmi
+endef
```

Comment:
why are these needed? how are these used?

Reply:
These targets are used for the usual operations but with respect to the 3 mocks (representing the catalog manager, credential manager, policy manager GRPC interfaces)
make build : builds the catalog connector and policy manager mock. 
make run, terminate : executes/terminates the catalog connector and policy manager mock. 
make dockerall : build mocks docker images for catalog connector and policy manager mock. 
make clean: cleans the mocks related logs
make docker-rmi: remove docker images for catalog connector and policy manager mock. 
make docker-push: push docker images for catalog connector and policy manager mock to the relevant docker registry

Also this makefile is integrated from the pkg/policy-compiler makefile as well so that all the connectors, mocks, external servers (like opa, egeria), are executed in sequence (in the scenario of local machine testing) and we can quickly check if there are any issues with the connectors execution logic (without need to deploy on the cluster env).